test/conformance/enqueue/urEnqueueUSMMemset2D.cpp

```diff
@@ -1,2 +1,215 @@
 // Copyright (C) 2023 Intel Corporation
 // SPDX-License-Identifier: MIT
+#include "helpers.h"
+#include <uur/fixtures.h>
+
+struct urEnqueueUSMMemset2DTestWithParam
+    : uur::urQueueTestWithParam<uur::TestParameters2D> {
+    void SetUp() override {
+        UUR_RETURN_ON_FATAL_FAILURE(urQueueTestWithParam::SetUp());
+
+        pitch = std::get<1>(GetParam()).pitch;
+        width = std::get<1>(GetParam()).width;
+        height = std::get<1>(GetParam()).height;
+        num_elements = pitch * height;
+        host_mem = std::vector<char>(num_elements);
+
+        bool device_usm{false};
+        ASSERT_SUCCESS(urDeviceGetInfo(device, UR_DEVICE_INFO_USM_DEVICE_SUPPORT,
+                                       sizeof(bool), &device_usm, nullptr));
+
+        if (!device_usm) {
+            GTEST_SKIP_("Device USM is not supported");
+        }
+
+        ur_usm_mem_flags_t flags{};
+        ASSERT_SUCCESS(urUSMDeviceAlloc(context, device, &flags, num_elements, 0,
+                                        reinterpret_cast<void **>(&ptr)));
+        ASSERT_SUCCESS(urQueueFlush(queue));
+    }
+
+    void TearDown() override {
+        if (ptr) {
+            EXPECT_SUCCESS(urUSMFree(context, ptr));
+        }
+
+        UUR_RETURN_ON_FATAL_FAILURE(urQueueTestWithParam::TearDown());
+    }
+
+    bool verifyData() {
+        EXPECT_SUCCESS(
+            urEnqueueUSMMemcpy2D(queue, true, host_mem.data(), pitch, ptr, pitch,
+                                 width, height, 0, nullptr, nullptr));
+        for (int w = 0; w < width; ++w) {
```

Comment:
Do you need a flush here?

Reply:
I don't think it's needed. urEnqueueUSMMemcpy2D is blocking. So I would assume the implementation does the flush.

Reply:
👍